octoprint_firmwareupdate/__init__.py

```diff
@@ -1,85 +1,271 @@
 # coding=utf-8
 from __future__ import absolute_import
 
-__author__ = "Kevin Murphy <kevin@voxel8.co>"
-__license__ = "GNU Affero General Public License http://www.gnu.org/licenses/agpl.html"
-__copyright__ = "Copyright (C) 2015 Kevin Murphy - Released under terms of the AGPLv3 License"
-
 import octoprint.plugin
-from octoprint.util import RepeatedTimer
 import flask
 import os
-from subprocess import Popen, call
-import signal
-import linecache
+from subprocess import Popen
 import psutil
-import time
+from time import sleep
+import requests
+import urllib
+from threading import Thread
+from glob import glob
+from serial import Serial, SerialException
+from octoprint.events import eventManager, Events
+
+Events.FIRMWARE_UPDATE = "FirmwareUpdate"
+
+__author__ = "Kevin Murphy <kevin@voxel8.co>"
+__license__ = "GNU Affero General Public License http://www.gnu.org/licenses/agpl.html"
+__copyright__ = "Copyright (C) 2015 Kevin Murphy - Released under terms of the AGPLv3 License"
+
 
 class FirmwareUpdatePlugin(octoprint.plugin.StartupPlugin,
-                       octoprint.plugin.TemplatePlugin,
-                       octoprint.plugin.AssetPlugin,
-                       octoprint.plugin.SettingsPlugin,
-                       octoprint.plugin.SimpleApiPlugin):
+                           octoprint.plugin.TemplatePlugin,
+                           octoprint.plugin.AssetPlugin,
+                           octoprint.plugin.SettingsPlugin,
+                           octoprint.plugin.SimpleApiPlugin):
 
     def __init__(self):
         self.isUpdating = False
```

Comment:
Can you document the important instance variables here? (A one line comment above them would be fine)